dlt/extract/pipe.py

```diff
@@ -588,6 +591,8 @@ def _get_source_item(self) -> ResolvablePipeItem:
         # TODO: count items coming of the generator and stop the generator if reached. that allows for sampling the beginning of a stream
         # _counts(id(gen)).setdefault(0) + 1
         try:
+            global _CURRENT_PIPE
```

Comment:
this will unfortunately not work with async and thread pool :/ my take for now: require that `resource_state` gets explicit resource name. this we can have in incremental. and we do not expose the `resource_state` via dlt.current for now until this is solved

Reply:
Ah ok. Is it more complicated than just multiple of these generators running concurrently? A contextvar instead of global would not be enough?